src/common/nodelist-utils/index.js

```diff
@@ -0,0 +1,16 @@
+// returns first node in list with matching first character (case-insenstive)
+function findNodeWithFirstChar(nodeList, char) {
+    // e.g. <li>A</li><li>B</li><li></li><li>B</li><li>A</li>
+
+    const keyValuePairList = [...nodeList].map((el, index) => ({ [el.innerText.charAt(0).toLowerCase()]: index }));
```

Comment:
Not entirely sure what you mean. Remember though it needs to return the index position of the node. Not sure if that is apparent. So long as it fullfils this contract I'm okay with the internals as they are for now, unless I'm doing something very, very silly... which is why I suck at code interviews ;-)

Reply:
I think it might be cleaner to do this (that is if it does the same thing)
return [...nodeList].findIndex((el) => el.innerText.charAt(0).toLowerCase() === char);

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
Thanks Andrew! Good call! I'd forgotten about `findIndex` (so many goodies in JavaScript these days). It doesn't quite work for the empty inner text scenario (the tests fail), but I'll see if I can rejig things and make use of `findIndex`.

Reply:
👍

Reply:
Ah silly me, just needed to change tests to check for `-1` instead of `undefined`. MUCH simpler now. Thanks.

Reply:
It's so simple I could potentially just remove the utility and tests altogether and just inline this.

Reply:
It does feel slightly overkill as a util method now, but I'm going to leave it in so that A) Listbox component (and maybe some others) can use it and B) we don't attempt to write it again in future in a long-winded fashion like I did :-)